services/token_update_nfts.proto

```diff
@@ -34,19 +34,28 @@ import "google/protobuf/wrappers.proto";
 /**
  * At consensus, updates an already created Non Fungible Token to the given values.
  * 
- * If no value is given for a field, that field is left unchanged. For an immutable tokens (that is,
- * a token without an admin key), only the expiry may be updated. Setting any other field in that
- * case will cause the transaction status to resolve to TOKEN_IS_IMMUTABLE.
+ * If no value is given for a field, that field is left unchanged.
+ * Only certain fields such as metadata can be updated.
  *
+ * Updating the metadata of an NFT does not affect its ownership or transferability.
+ * This operation is intended for updating attributes of individual NFTs in a collection.
+ *
+ * --- Signing Requirements ---
+ * 1. To update metadata of an NFT, the metadata_key of the token should sign the transaction.
  */
-message TokenUpdateNftTransactionBody {
+message TokenUpdateNftsTransactionBody {
+    /**
+     * The token for which to update NFTs.
+     */
+    TokenID token = 1;
+
     /**
-     * The NftID of the NFT
+     * The list of serial numbers to be updated.
      */
-    NftID nft_id = 1;
+    repeated int64 serial_numbers = 2;
 
     /**
-     * The new metadata of the NFT
+     * The new metadata of the NFT(s)
      */
-    google.protobuf.BytesValue metadata = 2;
-}
+    google.protobuf.BytesValue metadata = 3;
```

Comment:
Q: Should this be a list as well? in token mint, it's a list so newly mint nfts can have different metadata. I don't know how important it is to batch update multiple nft's metadata to be the same.

Reply:
yeah as per the [HIP](https://hips.hedera.com/hip/hip-657) this transaction updates _same_ `metadata` for a batch of NFTs. 
@netopyr thoughts?

Reply:
Right, the same metadata should be set for all NFTs.